onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQLinearConcat.cpp

```diff
@@ -0,0 +1,236 @@
+            0 // guaranteedBaseOffsetAlignment
+        );
+
+        // Validate input tensors
+        for (uint32_t input_index = 0; input_index < input_count; ++input_index)
+        {
+            // Inputs(input tensor, scale, zero_point) are in tuple and starting from index 2
+            auto tuple_start = 2 + input_index * 3;
+            auto xScaleDataType = kernelCreationContext.GetInputEdgeDescription(tuple_start + 1).tensorDataType;
+            auto xZeroPointDataType = kernelCreationContext.GetInputEdgeDescription(tuple_start + 2).tensorDataType;
+            ML_CHECK_VALID_ARGUMENT(xScaleDataType == yScaleDataType, "Wrong input type encountered for scale");
+            ML_CHECK_VALID_ARGUMENT(xZeroPointDataType == yZeroPointDataType, "Wrong input type encountered for zero point");
+
+            // broadcast x_scale and x_zero_point to shape of corresponding x
+            m_inputTensorDescs[tuple_start + 1] = TensorDesc(
+                kernelCreationContext.GetInputEdgeDescription(tuple_start + 1).tensorDataType,
+                kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(tuple_start),
+                kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(tuple_start + 1),
+                TensorAxis::DoNotCoerce,
+                TensorAxis::W,
+                TensorAxis::RightAligned,
+                NchwDimensionCount, // minDimensionCount
+                0 // guaranteedBaseOffsetAlignment
+            );
+
+            m_inputTensorDescs[tuple_start + 2] = TensorDesc(
+                kernelCreationContext.GetInputEdgeDescription(tuple_start + 2).tensorDataType,
+                kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(tuple_start),
+                kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(tuple_start + 2),
+                TensorAxis::DoNotCoerce,
+                TensorAxis::W,
+                TensorAxis::RightAligned,
+                NchwDimensionCount, // minDimensionCount
+                0 // guaranteedBaseOffsetAlignment
+            );
+        }
+
+        uint32_t dmlAxis = GetDmlAdjustedAxis(m_axis, kernelCreationContext, m_inputTensorDescs.front().GetDimensionCount(), 2);
+
+        std::vector<DML_TENSOR_DESC> inputDescs = GetDmlInputDescs();
+        std::vector<DML_TENSOR_DESC> outputDescs = GetDmlOutputDescs();
+
+        // 1. output edges between Dequantize and Join node
+        // 2. input edge between Join and Quantize node
+        std::vector<TensorDesc> intermediateOutputTensorDescs(input_count);
+        std::vector<DML_TENSOR_DESC> namedDequantizeOperatorDescs(input_count);
+        std::vector<DML_ELEMENT_WISE_DEQUANTIZE_LINEAR_OPERATOR_DESC> dequantizeOperatorDescs(input_count);
+        std::vector<DML_OPERATOR_DESC> dmlOpDesc(input_count);
+        std::vector<const DML_OPERATOR_DESC*> opDescs = {};
+        for (uint32_t input_index = 0; input_index < input_count; ++input_index)
```

Comment:
[nit] Is there any specific reason we have used `= {}` to initialize this particular std::vector?